Sources/SwiftPlot/Renderer.swift

```diff
@@ -193,3 +193,140 @@ extension Renderer {
         (self.xOffset, self.yOffset) = oldOffset
     }
 }
+
+/// Polygon structure definition and sequence extension, along with its iterator.
+public struct Polygon {
+    public var p1: Point, p2: Point, p3: Point
```

Comment:
By the polygon's `didSet` do you mean the wrapped array `didSet`? The idea was to force the programmer to always give at least 2/3 points at the point of creation so no check is needed or optionally using the failable initializer where the check happens once at runtime, but I agree that this is probably too much for what is essentially a wrapper.

Reply:
@karwa may I draw your attention to this? I will implement it as an array wrapper. Assuming you meant the `didSet` of the internal array of points, do we intend to allow people to just modify the internal array directly or should it be readonly? Maybe it makes sense to allow that, so these structs can start to be used earlier and points can be appended and such, but checking for the precondition for every single operation on the array may start hurting performance significantly eventually.

Reply:
Sure - I just mentioned the `didSet` because it is currently a `public var`, which allows mutations after the initialiser. Performance isn't really a massive concern - actually rendering the image will be the vast majority of the work. If it becomes a problem, we could make it `@inlinable` so the compiler can optimise those checks away. But really, only if it's a problem.